src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/RateAggregator.cs

```diff
@@ -58,25 +63,31 @@ public override IAggregationStatistics Collect()
                 double? delta = null;
                 if (_prevValue.HasValue)
                 {
-                    delta = _value - _prevValue.Value;
+                    delta = _currValue - _prevValue.Value;
                 }
-                RateStatistics stats = new RateStatistics(delta, _isMonotonic);
-                _prevValue = _value;
+                _aggregatedValue += _updated ? _currValue : 0;
```

Comment:
Ah here it is, I managed to look at this multiple times and still somehow overlook it. For an ObservableCounter the value reported in Update() is the absolute value of the counter at that instant in time. It should replace past values, rather than being summed with them.